src/c-writer.cc

```diff
@@ -3389,10 +3378,9 @@ void CWriter::Write(const ExprList& exprs) {
 
         Write(StackVar(0), " = (wasm_rt_funcref_t){", FuncTypeExpr(func_type),
               ", (wasm_rt_function_ptr_t)",
-              ExternalPtr(ModuleFieldType::Func, func->name), ", ");
+              ExternalRef(ModuleFieldType::Func, func->name), ", ");
```

Comment:
Can you explain this line?  What was ExternalPtr previously used and why is ExternalRef now OK to use instead?

Reply:
For functions, C lets us use the name of the function (instead of explicitly taking the address of the function) when we need a function pointer, which I think ends up cleaner in the output.

Here are some of the output diffs that show the changes:

- Calling an imported function:
  ```diff
  -  (*w2c_wasi__snapshot__preview1_proc_exit)(instance->w2c_wasi__snapshot__preview1_instance, var_i0);
  +  w2c_wasi__snapshot__preview1_proc_exit(instance->w2c_wasi__snapshot__preview1_instance, var_i0);
  ```

- Representing a funcref element segment:
  ```diff
   static const wasm_elem_segment_expr_t elem_segment_exprs_w2c_test_e0[] = {
  -  {w2c_test_t1, (wasm_rt_function_ptr_t)&w2c_test_f1, 0},
  +  {w2c_test_t1, (wasm_rt_function_ptr_t)w2c_test_f1, 0},
   };
  ```

Reply:
I see.. it wasn't obvious to me that that was the effect of this change, but that sgtm